h2/src/test/org/h2/test/db/TestIndex.java

```diff
@@ -155,20 +155,20 @@ private void testIndexTypes() throws SQLException {
     private void testErrorMessage() throws SQLException {
         reconnect();
         stat.execute("create table test(id int primary key, name varchar)");
-        testErrorMessage("PRIMARY", "KEY", " ON PUBLIC.TEST(ID)");
+        testErrorMessage("PRIMARY", "KEY", " ON \"\"PUBLIC\"\".\"\"TEST\"\"(\"\"ID\"\")");
```

Comment:
why are ending up with two quotes in a row here?

Reply:
The whole SQL in error messages is enclosed into double quotes. If it contains a double quote by itself it is replaced with two double quotes.

Reply:
```
Unique index or primary key violation: "PRIMARY KEY ON ""PUBLIC"".""TEST""(""ID"") [1, '1']"; SQL statement:
INSERT INTO TEST VALUES(1, 1) [23505-198]
```

Reply:
`MVPrimaryIndex` passes `PRIMARY KEY ON "PUBLIC"."TEST"("ID") [1, '1']` to `DbException.get()`.

`DbException.translate()` converts them to `""`.

Reply:
I guess that quoteIdentifier in DbException.translate() is not really useful anymore, after this change

Reply:
H2 passes different strings, including both quoted and unquoted identifiers, SQL code, values here. Perhaps all callers should be reviewed and normalized.